plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcModule.java

```diff
@@ -89,6 +89,15 @@ public void setup(Binder binder)
 
         newSetBinder(binder, ConnectorTableFunction.class);
 
+        install(conditionalModule(
+                QueryConfig.class,
+                QueryConfig::isRetryOpeningConnection,
+                innerBinder -> {
+                    innerBinder.bind(ConnectionFactory.class).annotatedWith(ForRetryJdbc.class).to(RetryingConnectionFactory.class);
+                    innerBinder.bind(RetryingConnectionCondition.class).toInstance(RetryingConnectionFactory::isRetryableException);
```

Comment:
I think `RetryingConnectionCondition` should be optional binder which would be overridden for Oracle connector.

So `RetryingConnectionFactory::isRetryableException` should be a default implemention of default `RetryingConnectionCondition`